### PR TITLE
Adding Apollo Link State Functionality (Draft)

### PR DESCRIPTION
**Pull request by sergei-gilevich**
In this pull request i added apollo-link-state to project. Also i removed redux dependency from counter module.

### CODE REVIEW

File: src/client/modules/connector.web.jsx [hunk 1]
```diff
@@ -68,6 +70,11 @@ export default class {
     return merge(...this.reducer);
   }
 
+  get resolvers() {
+    console.log('resolvers: ', merge(this.resolver)[0]);
```
Comment (vlasenko) [inline src/client/modules/connector.web.jsx]: Clean up debugging statements pls

File: src/client/modules/counter/components/CounterView.jsx [hunk 2]
```diff
@@ -23,10 +23,10 @@ const CounterView = ({ loading, counter, addCounter, reduxCount, onReduxIncremen
         <Button onPress={addCounter(1)}>Click to increase counter</Button>
         <View style={styles.element}>
           <Text style={styles.box}>
-            Current reduxCount, is {reduxCount}. This is being stored client-side with Redux.
+            Current reduxCount, is {counterState}. This is being stored client-side with Redux.
           </Text>
         </View>
-        <Button onPress={onReduxIncrement(1)}>Click to increase reduxCount</Button>
+        <Button onPress={addCounterState(1)}>Click to increase reduxCount</Button>
```
Comment (vlasenko) [inline src/client/modules/counter/components/CounterView.jsx]: Text should be updated, it is not Redux counter anymore

File: src/client/modules/counter/containers/Counter.jsx [hunk 3]
```diff
@@ -9,6 +8,8 @@ import CounterView from '../components/CounterView';
 import COUNTER_QUERY from '../graphql/CounterQuery.graphql';
 import ADD_COUNTER from '../graphql/AddCounter.graphql';
 import COUNTER_SUBSCRIPTION from '../graphql/CounterSubscription.graphql';
+import COUNTER_QUERY_CLIENT from '../graphql/CounterQuery_client.graphql';
```
Comment (vlasenko) [inline src/client/modules/counter/containers/Counter.jsx]: Why the convention change? I thought we have started to use `.client.graphql` in the projects, not `_client.graphql`, why this change?

File: src/client/modules/counter/containers/Counter.spec.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ import COUNTER_SUBSCRIPTION from '../graphql/CounterSubscription.graphql';
 chai.should();
 
 const COUNTER_APOLLO_VALUE = 11;
-const COUNTER_REDUX_VALUE = 7;
+const COUNTER_REDUX_VALUE = 1;
```
Comment (vlasenko) [inline src/client/modules/counter/containers/Counter.spec.js]: Why it is named REDUX here and why you change its value at all? Every change should have a reason.

File: packages/client/src/modules/counter/containers/Counter.spec.js [hunk 5]
```diff
@@ -63,7 +63,9 @@ describe('Counter example UI works', () => {
   step('Increase Redux counter button works', () => {
     const reduxButton = content.find('#redux-button').last();
     reduxButton.simulate('click');
-    content.text().should.has.string(`reduxCount, is ${COUNTER_REDUX_VALUE + 1}.`);
+    setTimeout(() => {
```
Comment (vlasenko) [inline packages/client/src/modules/counter/containers/Counter.spec.js]: This shouldn't be done like that, use `update` call instead

File: packages/client/src/modules/post/components/PostCommentsView.jsx [hunk 6]
```diff
@@ -13,57 +13,55 @@ export default class PostCommentsView extends React.PureComponent {
...
-    subscribeToMore: PropTypes.func.isRequired
+    subscribeToMore: PropTypes.func.isRequired,
+    addCommentClient: PropTypes.func.isRequired
   };
 
   keyExtractor = item => item.id;
 
   renderItem = ({ item: { id, content } }) => {
-    const { comment, deleteComment, onCommentSelect } = this.props;
+    const { comment, deleteComment, addCommentClient } = this.props;
```
Comment (vlasenko) [inline packages/client/src/modules/post/components/PostCommentsView.jsx]: Why to call the parameter `addCommentClient`, what does it means? Please use better name here

File: packages/client/src/modules/post/containers/Post.spec.js [hunk 7]
```diff
@@ -327,31 +328,29 @@ describe('Posts and comments example UI works', () => {
...
-
+    editButtons.last().simulate('click');
     const commentForm = content.find('form[name="comment"]');
     expect(
       commentForm
         .find('[name="content"]')
         .last()
         .instance().value
-    ).to.equal('Post comment 2');
+    ).to.equal('Post comment 2'); //must be Post comment 2
```
Comment (vlasenko) [inline packages/client/src/modules/post/containers/Post.spec.js]: This comment is redundant. It is evident that it must equal `Post comment 2`, comment doesn't add new information, so it is redundant.

File: packages/client/src/modules/post/containers/Post.spec.js [hunk 8]
```diff
@@ -327,18 +328,17 @@ describe('Posts and comments example UI works', () => {
...
       expect(input.content).to.equal('Edited comment 2');
       done();
       return input;
     };
-
     const editButtons = content.find('.edit-comment');
     expect(editButtons).has.lengthOf(6);
     editButtons.last().simulate('click');
-
+    editButtons.last().simulate('click');
```
Comment (vlasenko) [inline packages/client/src/modules/post/containers/Post.spec.js]: Why do we click two times here?

Comment (sergei-gilevich) [reply packages/client/src/modules/post/containers/Post.spec.js]: When I use a simulate('click') once, the test is fail. I tried several solutions from this article https://github.com/airbnb/enzyme/issues/1201, but they did not help.
I think this problem is related to the version of the "enzyme".

Comment (vlasenko) [reply packages/client/src/modules/post/containers/Post.spec.js]: @sergei-gilevich Oh, I see
